Detector/DetFCChhHCalTile/src/HCalBarrel_geo.cpp

```diff
@@ -42,14 +46,28 @@ static DD4hep::Geometry::Ref_t createHCal(DD4hep::Geometry::LCDD& lcdd, xml_h xm
   double dzSequence = sequenceDimensions.dz();
   lLog << MSG::DEBUG << "sequence thickness " << dzSequence << endmsg;
 
-  // calculate the number of modules fitting in phi, Z and Rho
+  // calculate the number of modules fitting in phi and Z
   unsigned int numSequencesPhi = sequenceDimensions.phiBins();
   double dphi = 2 * dd4hep::pi / static_cast<double>(numSequencesPhi);
-
   unsigned int numSequencesZ = static_cast<unsigned>((2 * dimensions.dz() - 2 * dZEndPlate - 2 * space) / dzSequence);
-  unsigned int numSequencesR = static_cast<unsigned>(
-      (cos(dphi / 2) * dimensions.rmax() - sensitiveBarrelRmin) /
-      sequenceDimensions.dr());  // needed due to trapezoid shapes inside the available HCAL volume
+
+  // Hard-coded assumption that we have three different layers for the modules
+  std::vector<xml_comp_t> Layers = {xmlElement.child(_Unicode(layer_1)), xmlElement.child(_Unicode(layer_2)), xmlElement.child(_Unicode(layer_3))};
+  unsigned int numSequencesR = 0;
+  double moduleDepth = 0.;
+  std::vector<double> layerDepths = std::vector<double>();
```

Comment:
I haven't hardcoded the number of layers, but the number of different types of layers.. I've corrected that in the comment..